Js/main.js

```diff
@@ -0,0 +1,13 @@
+const data =require('./data.js')
+// const searchInput=document.querySelector('.search-box')
+//Search By Name
+// searchInput.addEventListener('keypress', searchByName(searchInput.value))
+const searchByName =((search)=>{
+  const SearchedArray= data.filter((ele)=>{
```

Comment:
filter function actually returns the elements that pass the condition so you don't need an if statement here just pass the condition like this 
```   const SearchedArray= data.filter((ele)=> ele.name ===  search   ```